p2p/raintree/network.go

```diff
@@ -50,39 +52,64 @@ func NewRainTreeNetwork(addr cryptoPocket.Address, addrBook typesPre2P.AddrBook)
 		log.Println("[ERROR] Error initializing rainTreeNetwork: ", err)
 	}
 
-	return typesPre2P.Network(n)
+	return types2.Network(n)
 }
 
 func (n *rainTreeNetwork) NetworkBroadcast(data []byte) error {
 	return n.networkBroadcastAtLevel(data, n.maxNumLevels, getNonce())
 }
 
-func (n *rainTreeNetwork) networkBroadcastAtLevel(data []byte, level uint32, nonce uint64) error {
-	// This is handled either by the cleanup layer or redundancy layer
-	if level == 0 {
-		return nil
-	}
+func (n *rainTreeNetwork) networkBroadcastAtLevel(data []byte, level int32, nonce uint64) error {
+	var addr1, addr2 cryptoPocket.Address
+	var ok bool
 
-	msg := &typesPre2P.RainTreeMessage{
+	msg := &types2.RainTreeMessage{
 		Level: level,
 		Data:  data,
 		Nonce: nonce,
 	}
+	// This is handled either by the redundancy layer
+	if level == 0 {
```

Comment:
For clarity of code, can you move `redundancyLayer`  and `cleanupLayer` into separate functions. Similar to how `demote` is a separate function right now.